api/controllers/StatsController.js

```diff
@@ -6,6 +6,40 @@ dayjs.extend(isBetween);
 /////////////
 // Helpers //
 /////////////
+/**
+ * Find the season in which a given timestamp occurred using binary search
+ * @param {*} sortedSeasons - Seasons sorted (OLDEST FIRST)
+ * @param {*} timeStamp - Desired time in millis since epoch
+ * @returns the season | null if none is found
+ */
+function getSeasonFromTimeStamp(sortedSeasons, timeStamp) {
+  let currentIndex;
+  let minIndex = 0;
+  let maxIndex = sortedSeasons.length - 1;
+  // Sanity check in case list is unsorted or start & endTimes are out of whack
+  let numTries = 0;
+  while (numTries < sortedSeasons.length) {
```

Comment:
So this condition feels silly because so far as I can tell, there is actually no way that the while loop exits without returning `return currentSeason` or `return null`. I think this could actually have a constant condition, but it sketches me out to use `while (true)`.

It seems like a natural exit condition to use `while (minIndex < maxIndex)` except the way the loop is written atm we don't set the `currentIndex` or `currentSeason` until the beginning of the next loop (and each loop ends just with setting `minIndex` or `maxIndex`. I think that's relatively clean b/c each loop only winds up in one path and ultimately returns or sets one variable, but maybe it would make sense to set and check `currentIndex` + `currentSeason` at the end of the loop rather than the beginning so we could replace the block

```
    // If min = max and we haven't found it, there is no matching season
    if (minIndex === maxIndex) {
      return null;
    }
```

with an exit based on the while's condition using `while (minIndex < maxIndex)`

Reply:
I would love to talk through this more and see if there's a better data fetch we could do or a less flakey loop to get the data you're looking for.

Reply:
Is it possible to move the minIndex / maxIndex updating logic to the beginning of the loop and change the conditional to be `while (minIndex < maxIndex)`?

Reply:
> Is it possible to move the minIndex / maxIndex updating logic to the beginning of the loop and change the conditional to be `while (minIndex < maxIndex)`?

Yeah I think that would make more sense. It might have slightly more redundancy since atm the beginning of the first loop initialized currentIndex whereas I think we'd need to initialize it outside the loop to make this change but I think it would read more clearly if the exit condition were explicitly in the loop in that way, yeah

Reply:
I think I've worked out a reasonable compromise. @byarmis I think the conditional should be inclusive rather than exclusive:
```
while (minIndex <= maxIndex)
```

because if the updates to the `minIndex` and `maxIndex` come after the assignment and checks of `currentSeason`, then we need to do the exact same checks _before the loop_ in order to check the _first_ value of `currentSeason` (i.e. the halfway point of the list). By using `<=` instead of `<` and checking `currentSeason` at the beginning of the loop, but setting `currentIndex` at the _end_ of the loop, I think we get the best of both worlds since all the checking is still done inside the loop without needing to separately handle the first case/iteration, and we're still able to end the loop if the indices ever meet without finding the target season.